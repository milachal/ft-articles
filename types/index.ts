import { Dispatch, SetStateAction } from "react";

export interface HomeProps {
  articles: any;
}

export interface HomeArticleProps {
  meta: string,
  title: string
  standfirst: string,
  image: string,
}

export interface GetArticleResponse {
  meta: string,
  title: string
  standfirst: string,
  image: string,
  id: string,
}

export interface SuggestionsProps {
  suggestions: GetArticleResponse[],
  setShowSuggestions: Dispatch<SetStateAction<boolean>>
}

export interface SuggestionResult {
  apiUrl: string,
  aspectSet: string,
  id: string,
  modelVersion: string
}

export interface ArticlePageProps {
  article: {
    meta: string,
    title: string,
    standfirst: string,
    body: string,
    image: string,
    id: string
  }
}
