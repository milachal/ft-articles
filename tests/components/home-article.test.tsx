import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import HomeArticle from "../../components/home-article";
import { HomeArticleProps } from "../../types";

const initialProps: HomeArticleProps = {
  meta: "meta",
  title: "Some title",
  standfirst: "standfirst",
  image: "url"
};

let wrapper: ShallowWrapper;

beforeEach(() => {
  wrapper = shallow(
    <HomeArticle {...initialProps} />
  );
});

jest.mock("../../components/home-article/home-article.module.scss", () => <div />);

describe("Home article", () => {
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should test component content", () => {
    /* eslint-disable quotes */
    const meta = wrapper.find(`[data-testid="meta"]`);
    const title = wrapper.find(`[data-testid="title"]`);
    const standfirst = wrapper.find(`[data-testid="standfirst"]`);

    expect(meta.exists()).toBe(true);
    expect(meta.text()).toBe("meta");

    expect(title.exists()).toBe(true);
    expect(title.text()).toBe("Some title");

    expect(standfirst.exists()).toBe(true);
    expect(standfirst.text()).toBe("standfirst");
  });
});
