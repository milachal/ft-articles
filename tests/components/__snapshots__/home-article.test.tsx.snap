// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home article should render correctly 1`] = `
<div
  className="home-article-container"
>
  <img
    alt="article"
    className="home-article-container__image"
    src="url"
  />
  <div
    className="home-article-container__meta"
    data-testid="meta"
  >
    meta
  </div>
  <div
    className="home-article-container__title"
    data-testid="title"
  >
    Some title
  </div>
  <div
    className="home-article-container__standfirst"
    data-testid="standfirst"
  >
    standfirst
  </div>
</div>
`;
